chart/templates/postgres.yaml

```
@@ -52,26 +52,30 @@ spec:
 
 ---
 
-{{- if empty .Values.db.secretKeyRef.name }}
+{{- if .Values.db.secretKeyRef.create }}
 apiVersion: v1
 kind: Secret
 metadata:
-  name: postgres-connection
+  name: {{ .Values.db.secretKeyRef.name }}
   annotations:
     "helm.sh/resource-policy": "keep"
 type: Opaque
-data:
-  {{- $secretObj := (lookup "v1" "Secret" .Release.Namespace "postgres-connection") | default dict }}
-  {{ $releaseNamespace := .Release.Namespace }}
-  {{- $secretData := (get $secretObj "data") | default dict }}
-  {{- $user := (get $secretData "POSTGRES_USER") | default ("postgres" | b64enc) }}
+stringData:
+  {{- $secretObj := ( lookup "v1" "Secret" .Release.Namespace "postgres-connection" ) }}
+  {{- $secretData := ( get $secretObj "data" ) }}
+  {{- $user := (( get $secretData "POSTGRES_USER" ) | b64dec ) | default "postgres" }}
+  {{- $password := (( get $secretData "POSTGRES_PASSWORD" ) | b64dec ) | default randAlphaNum 32 }}
```

You're missing brackets, otherwise you get `Error: template: canary-checker/templates/postgres-secret.yaml:15:80: executing "canary-checker/templates/postgres-secret.yaml" at <randAlphaNum>: wrong number of args for randAlphaNum: want 1 got 0`

```suggestion
  {{- $password := (( get $secretData "POSTGRES_PASSWORD" ) | b64dec ) | default (randAlphaNum 32) }}
```